src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs

```diff
@@ -574,6 +580,7 @@ private void ReadImageHeaders(Stream stream, out bool inverted, out byte[] palet
                 this.stream.Read(palette, 0, colorMapSize);
             }
 
+            // TODO: ReSharper tells this expression is always true, looks like he's pretty right about it:
             if (this.infoHeader.Width > int.MaxValue || this.infoHeader.Height > int.MaxValue)
```

Comment:
You mean always false yeah? Been struggling to find a definitive answer but I think 65535x65535 is the maximum size on a 64 bit machine.

https://stackoverflow.com/questions/29175585/what-is-the-maximum-resolution-of-c-sharp-net-bitmap

Reply:
Always false of course :)

`65535 * 65535 = 4 294 836 225` and `int.MaxValue = 2 147 483 647` so `Width` and `Height` can't exceed that value in neither case.

`Width` and `Height` should be defined as `long` if we want to allow those sizes. (But I think we don't.)

Reply:
I think we're crossing wires here. `65535` is the max value for a `ushort` so not anywhere close to `int.MaxValue`. Incidentally that's the maximum length of any dimensions for a jpeg also.

Reply:
Omg true that, I should not make statements before lunch!

Maybe we should define something like an `Image.MaximumDimension` constant.

Reply:
I have thought about that before but it varies from format to format so we can't. 

I should print this out and frame it btw, I think it's the first time I've ever corrected you! 😝

Reply:
I think it already happened a few times! 😄
Fixed this stuff in my newest commit.